command/push.py

```diff
@@ -1,8 +1,32 @@
+import base64
+import os
+import requests
+import tarfile
+
 class PushCmd():
     name = 'push'
 
     def __init__(self):
         pass
         
-    def push(self):
-        print("Hello, world. The push feature is currently under construction.")
+    def push(self, bundle_dir, namespace, repository, release, auth_token):
+        base64_bundle = self.__create_base64_bundle(bundle_dir, repository)
+        self.__push_to_registry(namespace, repository, release, base64_bundle, auth_token)
+
+    def __create_base64_bundle(self, bundle_dir, repository):
+        tarfile_name = "%s.tar.gz" % repository
+        with tarfile.open(tarfile_name, "w:gz") as tar:
+            tar.add(bundle_dir, arcname=os.path.basename(bundle_dir))
+        with open(tarfile_name, "rb") as tarball:
+            result = tarball.read()
+        result64 = base64.b64encode(result).decode("utf-8")
+        return result64
+
+    def __push_to_registry(self, namespace, repository, release, bundle, auth_token):
```

Comment:
could you please use only one underscore, working with "private" methods in python is quite pain.